prusti-interface/src/environment/mir_utils/ty_as_ref.rs

```diff
@@ -0,0 +1,18 @@
+use rustc_middle::mir::terminator::Mutability;
+use rustc_middle::ty::Region;
+use rustc_middle::ty::Ty;
+use rustc_middle::ty::TyKind;
+
+pub trait TyAsRef<'tcx> {
+    fn as_ty_ref(&self) -> Option<(Region<'tcx>, Ty<'tcx>, Mutability)>;
```

Comment:
Why the function name does not match the file name?

Reply:
I renamed the file to `ty_as_ty_ref.rs`. The file name should convey that a `ty::Ty` is turned into a `ty::TyKind::Ref`. The function name should convey that the receiver (which is a `ty::Ty`) is turned into a `ty::TyKind::Ref`.